### PR TITLE
Stream Management - Perform bulk actions (delete)

### LINKED ISSUE #190: Add "bulk actions" to stream management
Editing streams can currently only be done one stream at a time. It would be nice to have a checkbox next to every stream in the admin tab and be able to apply bulk actions. 
E.g. or now: select multiple streams and click on a `delete selected` button instead of deleting them one by one.

**Todo:**
- [ ] #194
- [ ] Update titles
- [ ] Change lecture hall
- [ ] change start/end time
- [ ] move to day of week

@mono424 this might be something for now because streaming on the test server is not yet possible :)

### PR DESCRIPTION
**Pull request by mono424**
Enabled selecting multiple lectures/videos to delete them all at once.

### CODE REVIEW

**Comment by joschahenningsen** [comment]
Looks good, surprisingly little code as well 🥳 thanks! 
Can we make the delete lectures endpoint accept an array of ids in it's body instead of just one id from the url? Then we'd avoid spamming a bunch of consecutive requests for essentially one job.
Also I would expect the checkboxes to be on the left hand side, but that's just a matter of taste.

**Comment by mono424** [comment]
Actually thought already about both, endpoint makes totally sense! I will add that! Further I like the checkboxes aligned to the delete button as you can see better they are related, but can also move them to the left as you wish :)

**Comment by mono424** [comment]
Isn't. it wired if the checkbox is somehow directly below the filter checkbox even though they have no link? 
Bildschirmfoto 2021-12-21 um 12 34 43

File: api/courses.go [hunk 1]
```diff
@@ -353,21 +353,42 @@ type renameLectureRequest struct {
...
+		stream, err := dao.GetStreamByID(context.Background(), streamID)
+		if err != nil || stream.CourseID != tumLiveContext.Course.ID {
+			c.AbortWithStatus(http.StatusForbidden)
+			return
+		}
+		streams = append(streams, stream)
+	}
+
+	for _, stream := range streams {
+		stream.Model.DeletedAt = gorm.DeletedAt{Time: time.Now()} // todo ?!
```
Comment (joschahenningsen) [inline api/courses.go]: You don't need to set this, that's what gorm internally does when you call `Delete()`

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
This looks great :+1: one bulk request for a bulk actions seems like the way to go :)
How about moving the filter to the right and the bulk action (checkbox and button) to the left? 

But other than the comment on gorm, this looks good!

File: api/courses.go (hunk 1 above)
Comment (mono424) [reply api/courses.go]: Nice thanks, I will make the changes ASAP! :)

Comment (mono424) [reply api/courses.go]: What do you think now ? :)
Bildschirmfoto 2021-12-31 um 16 48 37

**Comment by joschahenningsen** [comment]
Thanks, this looks good :) Can you just roll back the diff in `package-lock.json` (and ideally use npm 8.x)?

**Comment by mono424** [comment]
> Thanks, this looks good :) Can you just roll back the diff in `package-lock.json` (and ideally use npm 8.x)?

Done, also updated npm locally to 8.3 :)

**Comment by joschahenningsen** [comment]
Great! Thanks a lot :fire: 

How about adding a "change lecture-hall" bulk action next? :)

**Review by joschahenningsen — APPROVED** [review]

**Pull request merged**